src/geo/geometry-views/leaflet/leaflet-marker-adapter.js

```diff
@@ -41,10 +40,11 @@ LeafletMarkerAdapter.prototype.getIconURL = function () {
 };
 
 LeafletMarkerAdapter.prototype.setIconURL = function (iconURL) {
-  var newIcon = L.icon({
-    iconUrl: iconURL
-  });
-  this._nativeMarker.setIcon(newIcon);
+  // Leaflet provides the `setIcon` method which generates a new img for the icon.
+  // We want to be able to update the markers icon while it's being dragged. That's
+  // why we're doing this little hack that changes the src of the existing marker img.
+  this._nativeMarker._icon.src = iconURL;
```

Comment:
We are removing an img node to place another one in his place, the new node doesn't have the mousedown event fired, so it doesn't add the drag, drag start, etc...

Reusing the img node we are avoiding some extra work from the browser and also reusing all the attached events so it works perfectly.

Why is this happening now? It is hard to give up with an answer without make some deep debuging work over Leaflet, they changed the way their events works.